pkg/controller/utils/velero.go

```diff
@@ -0,0 +1,42 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	velerov1 "github.com/heptio/velero/pkg/apis/velero/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	// DefaultExcludedBackupResources is the deault list of excludes when backing up resources.
+	DefaultExcludedBackupResources = []string{
+		// NOTE: even though pods is in the "core" group, specifying "pods.core" here will cause
+		// pods to BE BACKED UP (not excluded)!!!
+		"pods",
+		"jobs.batch",
+	}
+)
+
+// CreateVeleroBackupObjectForNamespace creates a Velero Backup object for the namespace specified.
+// The Backup options are set specifically for Hive object backups.
+// DO NOT use this function call for any other type objects as it may not back them up correctly.
+func CreateVeleroBackupObjectForNamespace(client client.Client, namespace string) error {
```

Comment:
IMO this function should not be in ` pkg/controller/utils/`. This is perfect for keeping in `pkg/controller/backup`. Let me know if I am missing anything.

Reply:
I agree that `pkg/controller/utils` is not a good place for this. I suggest either keep it in `pkg/controller/backup` or, if it is something not tied to controller behavior, then put it in `pkg/backup`. As it is, this file and its own function are small enough to make it hard to justify its own distinct package.